src/nodes.coffee

```diff
@@ -760,6 +760,45 @@ exports.IdentifierLiteral = class IdentifierLiteral extends Literal
   eachName: (iterator) ->
     iterator @
 
+exports.SubStructAssign = class SubStructAssign extends Base
+  constructor: (@variable, @value, @lhs = no) ->
+    super()
+    [@objVar, @objVal] = @variable
+
+  objProperties: (props) ->
+    ret = []
+    for prop in props
+      if prop instanceof Assign
+        ret.push prop.value.base
+      else if prop instanceof Splat
+        ret.push prop.name
+      else if prop.base not instanceof ExProperty
+        ret.push prop.base
+    ret
+
+  finalProps: (props = []) ->
+    ret = []
+    getProps = (props) => (p for p in @finalProps @objProperties props)
```

Comment:
why the for/in here? it's not here L792

Reply:
Right. I forgot to remove it. It's a leftover from the previous version.